packages/core/src/sdk/geolocation/useGeolocation.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react'
+
+import { deliveryPromise } from 'discovery.config'
+import { useSession, validateSession, sessionStore } from 'src/sdk/session'
+
+export default function useGeolocation() {
+  const { isValidating: _, ...session } = useSession()
+
+  useEffect(() => {
+    if (!deliveryPromise.enabled) {
+      return
+    }
+
+    if (navigator?.geolocation && !session.geoCoordinates) {
+      navigator.geolocation.getCurrentPosition(
+        async ({ coords: { latitude, longitude } }) => {
+          const newSession = {
+            ...session,
+            geoCoordinates: { latitude, longitude },
+          }
+          const validatedSession = await validateSession(newSession)
+          sessionStore.set(validatedSession ?? newSession)
+        }
+      )
+    }
+  }, [])
```

Comment:
This array should not have some dependencies?

Reply:
No, this consent is called just once because we should not ignore users' deny/dismiss and ask again for consent.
In fact, the Geolocation API itself warns us for that:
![Screenshot 2025-03-21 at 11 18 11](https://github.com/user-attachments/assets/7ca48bd3-297e-4af9-bace-9dc898c8a24e)